deeppavlov/models/torch_bert/torch_transformers_squad.py

```diff
@@ -177,65 +177,91 @@ def __call__(self, features: List[InputFeatures]) -> Tuple[List[int], List[int],
             predictions: start, end positions, start, end logits positions
 
         """
-        input_ids = [f.input_ids for f in features]
-        input_masks = [f.attention_mask for f in features]
-        input_type_ids = [f.token_type_ids for f in features]
-
-        b_input_ids = torch.cat(input_ids, dim=0).to(self.device)
-        b_input_masks = torch.cat(input_masks, dim=0).to(self.device)
-        b_input_type_ids = torch.cat(input_type_ids, dim=0).to(self.device)
-        
-        input_ = {
-            'input_ids': b_input_ids,
-            'attention_mask': b_input_masks,
-            'token_type_ids': b_input_type_ids,
-            'return_dict': True
-        }
-
-        with torch.no_grad():
-            input_ = {arg_name: arg_value for arg_name, arg_value in input_.items() if arg_name in self.accepted_keys}
-            # Forward pass, calculate logit predictions
-            outputs = self.model(**input_)
-
-            logits_st = outputs.start_logits
-            logits_end = outputs.end_logits
-
-            bs = b_input_ids.size()[0]
-            seq_len = b_input_ids.size()[-1]
-            mask = torch.cat([torch.ones(bs, 1, dtype=torch.int32),
-                              torch.zeros(bs, seq_len - 1, dtype=torch.int32)], dim=-1).to(self.device)
-            logit_mask = b_input_type_ids + mask
-            logits_st = softmax_mask(logits_st, logit_mask)
-            logits_end = softmax_mask(logits_end, logit_mask)
-
-            start_probs = torch.nn.functional.softmax(logits_st, dim=-1)
-            end_probs = torch.nn.functional.softmax(logits_end, dim=-1)
-            scores = torch.tensor(1) - start_probs[:, 0] * end_probs[:, 0]  # ok
-
-            outer = torch.matmul(start_probs.view(*start_probs.size(), 1),
-                                 end_probs.view(end_probs.size()[0], 1, end_probs.size()[1]))
-            outer_logits = torch.exp(logits_st.view(*logits_st.size(), 1) + logits_end.view(
-                logits_end.size()[0], 1, logits_end.size()[1]))
-
-            context_max_len = torch.max(torch.sum(b_input_type_ids, dim=1)).to(torch.int64)
-
-            max_ans_length = torch.min(torch.tensor(20).to(self.device), context_max_len).to(torch.int64).item()
-
-            outer = torch.triu(outer, diagonal=0) - torch.triu(outer, diagonal=outer.size()[1] - max_ans_length)
-            outer_logits = torch.triu(outer_logits, diagonal=0) - torch.triu(
-                outer_logits, diagonal=outer_logits.size()[1] - max_ans_length)
-
-            start_pred = torch.argmax(torch.max(outer, dim=2)[0], dim=1)
-            end_pred = torch.argmax(torch.max(outer, dim=1)[0], dim=1)
-            logits = torch.max(torch.max(outer_logits, dim=2)[0], dim=1)[0]
+        predictions = {}
```

Comment:
```suggestion
        # TODO: refactor batchification
        predictions = {}
```
Since this code works, I don't see any problem to keep it as it is, but it could be optimized so add todo